### PR TITLE
feat: case & namespace file exchange (frontend)

### PR DESCRIPTION
**Pull request by selimbensenouciep-prog**
## What

Adds the AgentOS file-exchange UI: a drawer that lists and manages the files shared within a case and across a namespace, plus the agent-config wiring to enable it.

## UI

* **Exchange drawer** (shell + content viewer): lists case files (read/write) and namespace files (read-only), with upload, download, delete, and an inline content/preview viewer.
* **ExchangeStateService** (root singleton): per-scope manifests exposed as signals, fail-closed capability mapping (a 403 or 404 hides the section, an error never grants a write affordance), and **live refresh** when the agent creates or deletes a file during a turn. Refresh is per file operation, with an end-of-turn safety net that only fires when a tool actually ran (pure-conversation turns do not refetch). The agent only mutates the case scope, so the read-only namespace manifest is never refetched on agent activity.
* **Agent config form**: a "Built-in integrations" section that lists case and namespace file exchange at the same level as the other integrations, discovered dynamically from the backend (no persisted booleans), and shown only when the file plugin is available.

## Design system

New shared components used by the drawer:

* `ds-confirm-dialog` (delete confirmation)
* `ds-empty-state` (empty file lists)
* `ds-spinner` (loading)
* a `position` input on `ds-drawer` (left / right)

## API client

Regenerated `agentos-api-client` from the backend OpenAPI for the new exchange endpoints and the built-in integration descriptors.

## Notes

* Branched from the delegation branch (the PR base), so this diff is the file-exchange work only.
* Requires the matching backend PR (`feat/selim.bensenouci/file-exchange-back`) at runtime for the endpoints. The storage layout, configuration parameters, and upload size limits are documented there.
* `pnpm-lock.yaml` is intentionally left out of this PR.

### CODE REVIEW

File: libs/agentos-ui/src/lib/components/agent-config-form/agent-config-form.component.ts [hunk 1]
```diff
@@ -189,16 +216,30 @@ export class AgentConfigFormComponent implements OnInit {
...
 
+  /**
+   * Resolve whether the file-exchange plugin is loaded by looking for a `FILE_ACCESS`
+   * integration type. Resilient: any error (e.g. endpoint unavailable) resolves to false
+   * so it never blocks the integrations load — and the checkboxes stay hidden (fail-safe).
+   */
+  private loadBuiltInTypes(): Observable<IntegrationTypeDescriptor[]> {
+    return this.integrationTypeController.listTypesIntegrationType().pipe(
+      map((types) => types.filter((t) => t.builtIn === true)),
+      catchError(() => of([]))
```
Comment (mathieu-dearmey) [inline libs/agentos-ui/src/lib/components/agent-config-form/agent-config-form.component.ts]: at least log error?

File: libs/agentos-ui/src/lib/components/case-chat/case-chat.component.ts [hunk 2]
```diff
@@ -570,6 +599,15 @@ export class CaseChatComponent implements OnInit, OnDestroy {
...
   }
 
+  /** Whether any tool ran this turn — gates the end-of-turn exchange refresh (skips pure-chat turns). */
+  private anyToolResponseThisTurn = false
+
+  /** The exchange scope a mutating file-plugin tool (create/edit/remove/move) acts on, or null. */
+  private exchangeMutationScope(toolName?: string): 'case' | 'namespace' | null {
+    const match = /^(case|namespace)-exchange__(editFiles|remove|moveFile)$/.exec(toolName ?? '')
+    return match ? (match[1] as 'case' | 'namespace') : null
+  }
```
Comment (mathieu-dearmey) [inline libs/agentos-ui/src/lib/components/case-chat/case-chat.component.ts]: test?

File: libs/agentos-ui/src/lib/components/exchange-shell/exchange-shell.component.ts [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+        // param must not wipe the open file and double-refetch both manifests.
+        if (caseId === this.activeCaseId) return
+        this.activeCaseId = caseId
+        this.resetSelection()
+        this.state.initializeForCase(namespaceId, caseId)
+      } else if (this.activeCaseId !== null) {
+        this.activeCaseId = null
+        this.state.clear()
+      }
+    })
```
Comment (mathieu-dearmey) [inline libs/agentos-ui/src/lib/components/exchange-shell/exchange-shell.component.ts]: can be done directly in constructor. No destroyRef needed in takeUntilDestroyed when in injection context of constructor

File: libs/agentos-ui/src/lib/components/exchange-shell/exchange-shell.component.ts [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+  }
+
+  ngOnDestroy(): void {
+    this.state.clear()
+  }
```
Comment (mathieu-dearmey) [inline libs/agentos-ui/src/lib/components/exchange-shell/exchange-shell.component.ts]: can be done in constructor > effect > onCleanup callback

File: libs/design-system/src/lib/components/confirm-dialog/confirm-dialog.component.ts [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+  EventEmitter,
+  Input,
+  OnChanges,
+  Output,
+  SimpleChanges,
+  ViewChild,
+} from '@angular/core'
+
+/** Per-instance counter so the message/title ids are unique across concurrently-rendered dialogs. */
+let dialogInstanceCounter = 0
```
Comment (mathieu-dearmey) [inline libs/design-system/src/lib/components/confirm-dialog/confirm-dialog.component.ts]: module-level mutable counter for a unique ID feels a bit heavy. crypto.randomUUID()  in the constructor would do the same without shared state

File: libs/design-system/src/lib/components/spinner/spinner.component.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  templateUrl: './spinner.component.html',
+  styleUrl: './spinner.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SpinnerComponent {
+  /** Visual size of the spinner. */
+  @Input() size: SpinnerSize = 'md'
+
+  /** Accessible label announced to assistive technology. */
+  @Input() label: string = 'Loading'
```
Comment (mathieu-dearmey) [inline libs/design-system/src/lib/components/spinner/spinner.component.ts]: input signals please

File: libs/design-system/src/lib/components/confirm-dialog/confirm-dialog.component.ts [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+  @Input() cancelLabel: string = 'Cancel'
+
+  /** Whether the dialog is visible. */
+  @Input() open: boolean = false
+
+  /** Emitted when the user confirms the action. */
+  @Output() confirmed = new EventEmitter<void>()
+
+  /** Emitted when the user cancels (button, Esc or backdrop click). */
+  @Output() cancelled = new EventEmitter<void>()
```
Comment (mathieu-dearmey) [inline libs/design-system/src/lib/components/confirm-dialog/confirm-dialog.component.ts]: input / output signals. please

File: libs/design-system/src/lib/components/confirm-dialog/confirm-dialog.component.ts [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+
+  @ViewChild('confirmButton') private confirmButton?: ElementRef<HTMLButtonElement>
```
Comment (mathieu-dearmey) [inline libs/design-system/src/lib/components/confirm-dialog/confirm-dialog.component.ts]: signal viewChild 🙏

File: libs/design-system/src/lib/components/confirm-dialog/confirm-dialog.component.ts [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+  /** Stable id for the message paragraph, referenced by the dialog's aria-describedby. */
+  protected readonly messageId = `ds-confirm-dialog-message-${dialogInstanceCounter++}`
+
+  private pendingFocus = false
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes['open'] && this.open) {
+      // Defer the focus to ngAfterViewChecked: the button only exists once @if renders it.
+      this.pendingFocus = true
+    }
```
Comment (mathieu-dearmey) [inline libs/design-system/src/lib/components/confirm-dialog/confirm-dialog.component.ts]: use computed on input signals

File: libs/design-system/src/lib/components/drawer/drawer.component.ts [hunk 10]
```diff
@@ -51,9 +51,18 @@ export class DrawerComponent implements OnInit {
...
+  @Input() position: 'start' | 'end' = 'start'
+
   /** Width of the sidenav panel. */
   @Input() drawerWidth: string = '280px'
 
+  /**
+   * When true, the panel spans the full viewport width (100vw) in mobile overlay mode,
+   * where a fixed `drawerWidth` would be too wide. Opt-in so existing drawers are unaffected.
+   */
+  @Input() mobileFullWidth: boolean = false
```
Comment (mathieu-dearmey) [inline libs/design-system/src/lib/components/drawer/drawer.component.ts]: input signals

File: libs/design-system/src/lib/components/empty-state/empty-state.component.ts [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+})
+export class EmptyStateComponent {
+  /** Optional Material icon name rendered above the title. */
+  @Input() icon?: string
+
+  /** Primary message describing the empty state. */
+  @Input() title: string = ''
+
+  /** Optional secondary text giving more context or next steps. */
+  @Input() hint?: string
```
Comment (mathieu-dearmey) [inline libs/design-system/src/lib/components/empty-state/empty-state.component.ts]: input signals

**Review by vincent-audibert-whoz — DISMISSED** [review]
Tested ok, except on the remove file with a simple agent, tool issue and not an exchange issue.

File: libs/agentos-ui/src/lib/components/agent-config-form/agent-config-form.component.ts (hunk 1 above)
Comment (selimbensenouciep-prog) [reply libs/agentos-ui/src/lib/components/agent-config-form/agent-config-form.component.ts]: Added a console.error in the catchError before the [] fallback, so the fail-safe still hides the toggles but is diagnosable. Done in 9f6b3544.

File: libs/agentos-ui/src/lib/components/case-chat/case-chat.component.ts (hunk 2 above)
Comment (selimbensenouciep-prog) [reply libs/agentos-ui/src/lib/components/case-chat/case-chat.component.ts]: Extracted exchangeMutationScope to exchange-content.utils.ts and added a covering unit test (mutating case/namespace tools -> scope; read-only/unknown/empty/undefined -> null). Done in 9f6b3544.

File: libs/agentos-ui/src/lib/components/exchange-shell/exchange-shell.component.ts (hunk 3 above)
Comment (selimbensenouciep-prog) [reply libs/agentos-ui/src/lib/components/exchange-shell/exchange-shell.component.ts]: Moved the query-param subscription into the constructor, so takeUntilDestroyed() no longer needs an explicit DestroyRef. Done in 9f6b3544.

File: libs/agentos-ui/src/lib/components/exchange-shell/exchange-shell.component.ts (hunk 4 above)
Comment (selimbensenouciep-prog) [reply libs/agentos-ui/src/lib/components/exchange-shell/exchange-shell.component.ts]: Replaced ngOnDestroy with inject(DestroyRef).onDestroy(() => this.state.clear()) in the constructor, so the component no longer implements OnInit/OnDestroy. Done in 9f6b3544.

File: libs/design-system/src/lib/components/confirm-dialog/confirm-dialog.component.ts (hunk 5 above)
Comment (selimbensenouciep-prog) [reply libs/design-system/src/lib/components/confirm-dialog/confirm-dialog.component.ts]: Good call - replaced the module-level counter with crypto.randomUUID() at field init, no shared state. Done in 9f6b3544.

File: libs/design-system/src/lib/components/spinner/spinner.component.ts (hunk 6 above)
Comment (selimbensenouciep-prog) [reply libs/design-system/src/lib/components/spinner/spinner.component.ts]: Migrated to signal inputs (input()). Done in 9f6b3544.

File: libs/design-system/src/lib/components/confirm-dialog/confirm-dialog.component.ts (hunk 7 above)
Comment (selimbensenouciep-prog) [reply libs/design-system/src/lib/components/confirm-dialog/confirm-dialog.component.ts]: Migrated to signal I/O (input()/output()). Done in 9f6b3544.

File: libs/design-system/src/lib/components/confirm-dialog/confirm-dialog.component.ts (hunk 8 above)
Comment (selimbensenouciep-prog) [reply libs/design-system/src/lib/components/confirm-dialog/confirm-dialog.component.ts]: Migrated to viewChild() signal. Done in 9f6b3544.

File: libs/design-system/src/lib/components/confirm-dialog/confirm-dialog.component.ts (hunk 9 above)
Comment (selimbensenouciep-prog) [reply libs/design-system/src/lib/components/confirm-dialog/confirm-dialog.component.ts]: The open/focus logic is now an effect() on the signal inputs (open() + the confirmButton viewChild), replacing ngOnChanges + ngAfterViewChecked. Done in 9f6b3544.

File: libs/design-system/src/lib/components/drawer/drawer.component.ts (hunk 10 above)
Comment (selimbensenouciep-prog) [reply libs/design-system/src/lib/components/drawer/drawer.component.ts]: Migrated to signal I/O; open is a model() for the [(open)] two-way binding (openChange auto-provided). Done in 9f6b3544.

File: libs/design-system/src/lib/components/empty-state/empty-state.component.ts (hunk 11 above)
Comment (selimbensenouciep-prog) [reply libs/design-system/src/lib/components/empty-state/empty-state.component.ts]: Migrated to signal inputs (input()). Done in 9f6b3544.

File: libs/agentos-ui/src/lib/components/exchange-shell/exchange-shell.component.ts [hunk 12]
```diff
@@ -0,0 +1,202 @@
...
+        this.activeCaseId = caseId
+        this.resetSelection()
+        this.state.initializeForCase(namespaceId, caseId)
+      } else if (this.activeCaseId !== null) {
+        this.activeCaseId = null
+        this.state.clear()
+      }
+    })
+    // Release the shared exchange state when this shell is torn down.
+    this.destroyRef.onDestroy(() => this.state.clear())
```
Comment (mathieu-dearmey) [inline libs/agentos-ui/src/lib/components/exchange-shell/exchange-shell.component.ts]: i had the following in mind
```suggestion
    effect((onCleanup) => onCleanup(() => this.state.clear()))
```
but I suppose `destroyRef.onDestroy` is just as good

**Review by mathieu-dearmey — DISMISSED** [review]

File: libs/agentos-ui/src/lib/components/exchange-shell/exchange-shell.component.ts (hunk 12 above)
Comment (selimbensenouciep-prog) [reply libs/agentos-ui/src/lib/components/exchange-shell/exchange-shell.component.ts]: Good call — switched to `effect((onCleanup) => onCleanup(() => this.state.clear()))` in 627376a8. Kept DestroyRef only for the other (method-scoped) content-load subscription, which isn't in an injection context.

**Review by mathieu-dearmey — APPROVED** [review]
